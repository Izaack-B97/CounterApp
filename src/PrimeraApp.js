import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
// import React from 'react';

// Funcional Component
const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <Fragment>
            <h1>{ saludo }!!!</h1>
            {/* <pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <p>{ subtitulo }</p>
        </Fragment>
        
    );

};

// Condiciones de las props 
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}




export default PrimeraApp;