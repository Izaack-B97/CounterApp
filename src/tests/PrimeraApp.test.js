import React from 'react';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp /> ', () => {

    // test('debe de mostrar el mensaje "Hola, soy Isaac"', () => {
    //     const saludo = 'Hola, soy Isaac';
    //     const wrapper = render( <PrimeraApp saludo={ saludo }/> );
    //     const { getByText } = wrapper;
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });
    

    test('should ', () => {
        const saludo = 'Hola, soy Isaac';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );

        expect( wrapper ).toMatchSnapshot();        

    })
    

});