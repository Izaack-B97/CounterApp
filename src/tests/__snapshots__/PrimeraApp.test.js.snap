// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp />  Debe mostrar reenderizar <PrimeraApp />  1`] = `
<Fragment>
  <h1>
    Hola, soy Isaac
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp />  should  1`] = `
<Fragment>
  <h1>
    Hola, soy Isaac
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
